Update Dockerfile_rocm64_whisperx

lsunay opened the pull request:
Hi Robert, 

First off, thank you for your work on the Dockerfile for Whisper-WebUI with ROCm (rocm64_gfx803_base). I've been using it and found it very helpful. 

I encountered an "exec format error" when running the image built with the Dockerfile. The error message was: exec /Whisper-WebUI/whisp.sh: exec format error. 

After some troubleshooting, I found that changing the ENTRYPOINT instruction in the Dockerfile resolved the issue for me. The container now starts correctly. 

This change explicitly invokes /bin/bash to execute the whisp.sh script, which seems to bypass a potential issue with the shebang line (#!/bin/bash) interpretation or line endings within the script itself when it's created via echo commands in the Dockerfile.

I wanted to share this in case it helps others who might encounter the same problem, or if you'd consider incorporating this change into the Dockerfile for broader compatibility. 

Thanks again for your efforts! 

Best regards,

robertrosenbusch commented:
@lsunay : Hi Levent and welcome back!

Relax. I didnt published the whisperX for ROCm 6.3.4/6.4 "official", cause i try out a completly new AMD ROCm Dockercontainer which cut off the used StorageSpace a lot meanwhile the Docker-Building-Process and to help on my self to maintain the different Projects(Ollama/ComfyUI/WhisperX). And of course to learn on my self.

And sometimes on the scripts i published i was wrong.

i guess the ROCm 6.3.4 script on WhisperX i published today will be make more sense, cause the prob was a simple additional ">" character to build the whisp.sh and start the Container without any probs:

`RUN touch whisp.sh && \
    echo '#!/bin/bash' > whisp.sh && \
    echo "MIOPEN_LOG_LEVEL=3 ./venv/bin/python3 app.py --whisper_type faster-whisper --share --server_name 0.0.0.0 --server_port ${WHISPERX_GUI_PORT}" >> whisp.sh && \
    chmod +x whisp.sh && \
    true   `

The time to build it on AMD 3700x with 48 Gigs and a 500 MBit fibre on HDD Spindle was around 1,5 hours ^.^:

`date&&docker build -f Dockerfile_rocm634_whisperx . -t 'rocm634_gfx803_whisperx:latest'&&date 
Di 6. Mai 21:35:02 CEST 2025
Endbuild:
Successfully tagged  
Di 6. Mai 23:08:48 CEST 2025`
----

@lsunay : In this way, thanks a lot for your hint and advice!
---
greatings Robert Rosenbusch

robertrosenbusch commented:
@lsunay Hi Levent and welcome back!

Thanx a lot of your feedback and relax. 

I didnt published the whisperX for ROCm 6.3.4/6.4 "official", cause i try out a completly new AMD ROCm Dockercontainer which cut off the used StorageSpace a lot meanwhile the Docker-Building-Process and to help on my self to maintain the different Projects(Ollama/ComfyUI/WhisperX). And of course to learn on my self.

And sometimes on the scripts i published are wrong on this new AMD ROCm stack^,^

i guess the ROCm 6.3.4 script on WhisperX i published today will be make more sense, cause the prob was a simple additional ">" character to build the whisp.sh and start the Container without any probs:

`RUN touch whisp.sh && \ echo '#!/bin/bash' > whisp.sh && \ echo "MIOPEN_LOG_LEVEL=3 ./venv/bin/python3 app.py --whisper_type faster-whisper --share --server_name 0.0.0.0 --server_port ${WHISPERX_GUI_PORT}" >> whisp.sh && \ chmod +x whisp.sh && \ true   
`
The time to build it on AMD 3700x with 48 Gigs and a 500 MBit fibre on HDD Spindle was around 1,5 hours ^.^:

`date&&docker build -f Dockerfile_rocm634_whisperx . -t 'rocm634_gfx803_whisperx:latest'&&date`  

Di 6. Mai 21:35:02 CEST 2025 Endbuild: Successfully tagged   Di 6. Mai 23:08:48 CEST 2025
@lsunay : In this way, thanks a lot for your hint and advice!

lsunay commented:
Subject: Clarification on "exec format error" fix & a robust ENTRYPOINT (Follow-up to #19) 

Hi Robert, 

Thanks again for your detailed response on issue #19 and for sharing your insights into the whisp.sh creation. 

I wanted to clarify my experience with the "exec format error." While ensuring the whisp.sh script uses >> for appending lines is definitely crucial (and I had corrected this in my local tests), I found that this step alone didn't fully resolve the exec format error on my end. 

The issue was consistently resolved only after I modified the ENTRYPOINT to explicitly use /bin/bash like this: 
dockerfile
 
 
 
1
ENTRYPOINT ["/bin/bash", "/Whisper-WebUI/whisp.sh"]
 
 

This leads me to believe that the exec format error I was encountering might have been related to subtle line ending issues (e.g., CRLF) being introduced into whisp.sh during its creation within the Docker build process, or a shebang interpretation nuance in the execution environment. Explicitly calling /bin/bash in the ENTRYPOINT seems to reliably handle these cases. 

Therefore, while a correctly formed whisp.sh (using > for the first line and >> for subsequent lines, with the correct #!/bin/bash shebang) is essential, I'd still recommend keeping the ENTRYPOINT as ["/bin/bash", "/Whisper-WebUI/whisp.sh"] for maximum robustness against potential exec format error issues that can be tricky to debug. It provides an extra layer of certainty. 

On a lighter note, I'm also on a steep learning curve with all this! I've successfully managed to build your images, but my journey has involved a significant investment of time and disk space through trial-and-error. It's been quite the adventure in "Docker-Space," and my storage thanks you for the workout! 😄 I'm glad I could contribute a small finding that might help others. 

Thanks for fostering a great learning environment and for your dedication to these ROCm projects. Your work is a huge help to the community. 

Best regards, 

Levent Sunay
